api/query/atoms.go

```diff
@@ -529,6 +551,35 @@ func (e *AbstractSequential) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// UnmarshalJSON for AbstractCount attempts to interpret a query atom as
+// {"count": int, "where": query}.
+func (c *AbstractCount) UnmarshalJSON(b []byte) error {
```

Comment:
what is the reasoning behind BindToRuns implements on a AbstractCount while UnmarshalJSON implements on *AbstractCount?

Reply:
You pass a ref if A) you're going to mutate the ref or B) it's costly to copy (a large struct).